diff --git a/protocols/jabber/libiris/src/irisnet/noncore/legacy/srvresolver.cpp b/protocols/jabber/libiris/src/irisnet/noncore/legacy/srvresolver.cpp
index 422402a0d5c89a3e2fc90b6e992398d0bb9ddbbd..0f9498a7459b34b65601fe0119aca1940a09897e 100644
--- a/protocols/jabber/libiris/src/irisnet/noncore/legacy/srvresolver.cpp
+++ b/protocols/jabber/libiris/src/irisnet/noncore/legacy/srvresolver.cpp
@@ -26,14 +26,14 @@
 
 // CS_NAMESPACE_BEGIN
 
-static void sortSRVList(QList<Q3Dns::Server> &list)
+static void sortSRVList(QList<QDns::Server> &list)
 {
-	QList<Q3Dns::Server> tmp = list;
+	QList<QDns::Server> tmp = list;
 	list.clear();
 
 	while(!tmp.isEmpty()) {
-		QList<Q3Dns::Server>::Iterator p = tmp.end();
-		for(QList<Q3Dns::Server>::Iterator it = tmp.begin(); it != tmp.end(); ++it) {
+		QList<QDns::Server>::Iterator p = tmp.end();
+		for(QList<QDns::Server>::Iterator it = tmp.begin(); it != tmp.end(); ++it) {
 			if(p == tmp.end())
 				p = it;
 			else {
@@ -76,7 +76,7 @@ public:
 
 	bool srvonly;
 	QString srv;
-	QList<Q3Dns::Server> servers;
+	QList<QDns::Server> servers;
 	bool aaaa;
 
 	QTimer t;
@@ -171,7 +171,7 @@ bool SrvResolver::isBusy() const
 		return false;
 }
 
-QList<Q3Dns::Server> SrvResolver::servers() const
+QList<QDns::Server> SrvResolver::servers() const
 {
 	return d->servers;
 }
@@ -211,10 +211,10 @@ void SrvResolver::nndns_resultsReady(const QList<XMPP::NameRecord> &results)
 
 	if(d->nntype == XMPP::NameRecord::Srv) {
 		// grab the server list and destroy the qdns object
-		QList<Q3Dns::Server> list;
+		QList<QDns::Server> list;
 		for(int n = 0; n < results.count(); ++n)
 		{
-			list += Q3Dns::Server(QString::fromLatin1(results[n].name()), results[n].priority(), results[n].weight(), results[n].port());
+			list += QDns::Server(QString::fromLatin1(results[n].name()), results[n].priority(), results[n].weight(), results[n].port());
 		}
 
 		d->nndns_busy = false;
diff --git a/protocols/jabber/libiris/src/irisnet/noncore/legacy/srvresolver.h b/protocols/jabber/libiris/src/irisnet/noncore/legacy/srvresolver.h
index 991716f3371b403306404b94cafa348c9c034ac0..ed6eb544f7007541fc08caf2bb280d596f02d74a 100644
--- a/protocols/jabber/libiris/src/irisnet/noncore/legacy/srvresolver.h
+++ b/protocols/jabber/libiris/src/irisnet/noncore/legacy/srvresolver.h
@@ -27,7 +27,7 @@
 
 // CS_NAMESPACE_BEGIN
 
-class Q3Dns
+class QDns
 {
 public:
 	class Server
@@ -56,7 +56,7 @@ public:
 	void stop();
 	bool isBusy() const;
 
-	QList<Q3Dns::Server> servers() const;
+	QList<QDns::Server> servers() const;
 
 	bool failed() const;
 	QHostAddress resultAddress() const;
