diff --git a/protocols/groupwise/gwconnector.cpp b/protocols/groupwise/gwconnector.cpp
index d1330ad824dc7c731dc37a95b8735273c4890a46..2c62eaa80249e5397110ec5ec83bd4a30333d9aa 100644
--- a/protocols/groupwise/gwconnector.cpp
+++ b/protocols/groupwise/gwconnector.cpp
@@ -18,9 +18,7 @@
  ***************************************************************************/
 
 #include "gwconnector.h"
-#include <k3bufferedsocket.h>
 #include <kdebug.h>
-#include <k3resolver.h>
 
 #include "gwerror.h"
 #include "gwbytestream.h"
diff --git a/protocols/groupwise/ui/gwchatprops.ui b/protocols/groupwise/ui/gwchatprops.ui
index f93fb5db8269323d5b28bd60181bbe3a209f80e3..5e987de01728ce4c99c30a556c6e602760581575 100644
--- a/protocols/groupwise/ui/gwchatprops.ui
+++ b/protocols/groupwise/ui/gwchatprops.ui
@@ -372,12 +372,12 @@
       <bool>false</bool>
      </property>
      <property name="buddy" >
-      <cstring>k3ListBox1</cstring>
+      <cstring>QListWidget1</cstring>
      </property>
     </widget>
    </item>
    <item>
-    <widget class="K3ListBox" name="acl" >
+    <widget class="QListWidget" name="acl" >
      <property name="toolTip" >
       <string>Access permissions for specific users</string>
      </property>
@@ -433,9 +433,9 @@
    <container>1</container>
   </customwidget>
   <customwidget>
-   <class>K3ListBox</class>
-   <extends>Q3ListBox</extends>
-   <header>k3listbox.h</header>
+   <class>QListWidget</class>
+   <extends>QListView</extends>
+   <header>QListWidget</header>
   </customwidget>
  </customwidgets>
  <tabstops>
@@ -457,7 +457,7 @@
   <tabstop>deleteAcl</tabstop>
  </tabstops>
  <includes>
-  <include location="local" >k3listbox.h</include>
+  <include location="local" >QListWidget</include>
   <include location="local" >kpushbutton.h</include>
   <include location="local" >kpushbutton.h</include>
   <include location="local" >kpushbutton.h</include>
