commit 283e3dcc90e0a1ebacfbeac0de5dfa8abc6cb6b5
Author: R. Harish Navnit <harishnavnit@gmail.com>
Date:   Sun Jun 14 19:20:06 2015 +0530

    Do away with k3listbox in texteffect plugin
    
    REVIEW: 124094

diff --git a/plugins/texteffect/texteffectpreferences.cpp b/plugins/texteffect/texteffectpreferences.cpp
index 35fac60d419cbac5644a8e143be1fc7c640385a9..856ca35b3345c6182dc3462aa1258817ff606dfe 100644
--- a/plugins/texteffect/texteffectpreferences.cpp
+++ b/plugins/texteffect/texteffectpreferences.cpp
@@ -22,6 +22,8 @@
 #include <qcheckbox.h>
 #include <qpushbutton.h>
 #include <QVBoxLayout>
+#include <QListWidget>
+#include <QListWidgetItem>
 
 #include <klocale.h>
 #include <kcolordialog.h>
@@ -96,7 +98,7 @@ void TextEffectPreferences::load()
 
 	config->load();
 
-	preferencesDialog->mColorsListBox->insertStringList(config->colors());
+	preferencesDialog->mColorsListBox->addItems(config->colors());
 	preferencesDialog->m_fg->setChecked(config->colorLines());
 	preferencesDialog->m_words->setChecked(config->colorWords());
 	preferencesDialog->m_char->setChecked(config->colorChar());
@@ -143,9 +145,9 @@ void TextEffectPreferences::save()
 QStringList TextEffectPreferences::colors()
 {
 	QStringList ret;
-	for(unsigned int f=0; f<preferencesDialog->mColorsListBox->count() ; f++)
+	for(int f=0; f < preferencesDialog->mColorsListBox->count() ; ++f)
 	{
-		ret.append(preferencesDialog->mColorsListBox->text(f));
+		ret.append(preferencesDialog->mColorsListBox->item(f)->text());
 	}
 	return ret;
 }
@@ -155,7 +157,7 @@ void TextEffectPreferences::slotAddPressed()
 	QColor myColor;
 	if( KColorDialog::getColor( myColor ) == KColorDialog::Accepted )
 	{
-		preferencesDialog->mColorsListBox->insertItem(myColor.name());
+		preferencesDialog->mColorsListBox->insertItem(preferencesDialog->mColorsListBox->currentRow(), myColor.name());
 	}
 
 	// Indicate that something has changed
@@ -164,7 +166,8 @@ void TextEffectPreferences::slotAddPressed()
 }
 void TextEffectPreferences::slotRemovePressed()
 {
-	delete preferencesDialog->mColorsListBox->selectedItem();
+	delete preferencesDialog->mColorsListBox->currentItem();
+
 	// Indicate that something has changed
 	slotSettingChanged();
 }
@@ -172,16 +175,17 @@ void TextEffectPreferences::slotRemovePressed()
 
 void TextEffectPreferences::slotUpPressed()
 {
-	int p=preferencesDialog->mColorsListBox->currentItem();
+	int p=preferencesDialog->mColorsListBox->currentRow();
 	if(p <= 0 )
 		return;
-	Q3ListBoxItem *i=preferencesDialog->mColorsListBox->selectedItem();
+
+	QListWidgetItem *i = preferencesDialog->mColorsListBox->currentItem();
 	if(!i)
 		return;
-	preferencesDialog->mColorsListBox->setSelected(i,false);
-	preferencesDialog->mColorsListBox->takeItem(i);
-	preferencesDialog->mColorsListBox->insertItem(i , p-1 );
-	preferencesDialog->mColorsListBox->setSelected(i,true);
+	i->setSelected(false);
+	preferencesDialog->mColorsListBox->takeItem(i->listWidget()->row(i));
+	preferencesDialog->mColorsListBox->insertItem(p-1, i);
+	i->setSelected(true);
 
 	// Indicate that something has changed
 	slotSettingChanged();
@@ -189,16 +193,17 @@ void TextEffectPreferences::slotUpPressed()
 }
 void TextEffectPreferences::slotDownPressed()
 {
-	int p=preferencesDialog->mColorsListBox->currentItem();
+	int p=preferencesDialog->mColorsListBox->currentRow();
 	if(p < 0 )
 		return;
-	Q3ListBoxItem *i=preferencesDialog->mColorsListBox->selectedItem();
+
+	QListWidgetItem *i = preferencesDialog->mColorsListBox->currentItem();
 	if(!i)
 		return;
-	preferencesDialog->mColorsListBox->setSelected(i,false);
-	preferencesDialog->mColorsListBox->takeItem(i);
-	preferencesDialog->mColorsListBox->insertItem(i , p+1 );
-	preferencesDialog->mColorsListBox->setSelected(i,true);
+	i->setSelected(false);
+	preferencesDialog->mColorsListBox->takeItem(i->listWidget()->row(i));
+	preferencesDialog->mColorsListBox->insertItem(p+1, i);
+	i->setSelected(true);
 
 	// Indicate that something has changed
 	slotSettingChanged();
@@ -217,7 +222,7 @@ void TextEffectPreferences::slotSettingChanged()
 void TextEffectPreferences::defaults()
 {
     preferencesDialog->mColorsListBox->clear();
-    preferencesDialog->mColorsListBox->insertStringList(config->defaultColorList());
+    preferencesDialog->mColorsListBox->addItems(config->defaultColorList());
     preferencesDialog->m_fg->setChecked(false);
     preferencesDialog->m_words->setChecked(false);
     preferencesDialog->m_char->setChecked(false);
diff --git a/plugins/texteffect/texteffectprefs.ui b/plugins/texteffect/texteffectprefs.ui
index 1b82f3de4b449373a3dedb39b664c0f645f1d02a..2d7f6a23cfa984c4f1ec7e1afa7f60580364c5d2 100644
--- a/plugins/texteffect/texteffectprefs.ui
+++ b/plugins/texteffect/texteffectprefs.ui
@@ -82,7 +82,7 @@
            <number>6</number>
           </property>
           <item rowspan="5" row="0" column="0" >
-           <widget class="K3ListBox" native="1" name="mColorsListBox" />
+			<widget class="QListWidget" native="1" name="mColorsListBox" />
           </item>
           <item row="0" column="1" >
            <widget class="QPushButton" name="mColorsAdd" >
@@ -239,12 +239,7 @@
    </item>
   </layout>
  </widget>
-  <customwidgets>
-  <customwidget>
-   <class>K3ListBox</class>
-   <extends></extends>
-   <header>k3listbox.h</header>
-  </customwidget>
+ <customwidgets>
  </customwidgets>
  <tabstops>
   <tabstop>TabWidget3</tabstop>
