diff --git a/libkopete/ui/addresseeitem.cpp b/libkopete/ui/addresseeitem.cpp
index e47f36c8c79e62ca593e282dc66b7874a12d789f..0d74e19a57f135772550bb2f91fbafeb43d5b660 100644
--- a/libkopete/ui/addresseeitem.cpp
+++ b/libkopete/ui/addresseeitem.cpp
@@ -21,15 +21,14 @@
 
 #include <qlayout.h>
 #include <qpushbutton.h>
-#include <q3groupbox.h>
 #include <qregexp.h>
 #include <QPixmap>
 
 #include <klocale.h>
 #include <kdebug.h>
 
-AddresseeItem::AddresseeItem( Q3ListView *parent, const KABC::Addressee &addressee) :
-  K3ListViewItem( parent ),
+AddresseeItem::AddresseeItem( QTreeWidget *parent, const KABC::Addressee &addressee) :
+  QTreeWidgetItem( parent ),
   mAddressee( addressee )
 {
   //We can't save showphoto because we don't have a d pointer
@@ -39,7 +38,7 @@ AddresseeItem::AddresseeItem( Q3ListView *parent, const KABC::Addressee &address
   if(pic.isIntern())
   {
     QPixmap qpixmap = QPixmap::fromImage( pic.data().scaledToWidth(60) ); //60 pixels seems okay.. kmail uses 60 btw
-    setPixmap( Photo,qpixmap );
+    //setPixmap( Photo,qpixmap );
   }
 
   setText( Name, addressee.realName() );
diff --git a/libkopete/ui/addresseeitem.h b/libkopete/ui/addresseeitem.h
index 080f770a1d3c316b8fbf4e4a7fc9d1c130f756fc..be47e4e80345844f823443a053c7c6470128ebf9 100644
--- a/libkopete/ui/addresseeitem.h
+++ b/libkopete/ui/addresseeitem.h
@@ -22,14 +22,14 @@
 #define ADDRESSEEITEM_H
 
 #include <klineedit.h>
-#include <k3listview.h>
+#include <QTreeWidget>
 
 #include <kabc/addressbook.h>
 
 /**
   @short Special ListViewItem
 */
-class AddresseeItem : public K3ListViewItem
+class AddresseeItem : public QTreeWidgetItem
 {
   public:
 
@@ -46,7 +46,7 @@ class AddresseeItem : public K3ListViewItem
       @param parent    The parent listview.
       @param addressee The associated addressee.
     */
-    AddresseeItem( Q3ListView *parent, const KABC::Addressee &addressee );
+    AddresseeItem( QTreeWidget *parent, const KABC::Addressee &addressee );
    
     /**
       Returns the addressee.
